amd/src/mappings.js

```diff
@@ -0,0 +1,155 @@
+// This file is part of Moodle - http://moodle.org/
+//
+// Moodle is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// Moodle is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Moodle.  If not, see <http://www.gnu.org/licenses/>.
+
+/*
+ * The module provides a function to refresh the user list with correct credential data.
+ *
+ * @module    mod_accredible
+ * @package   accredible
+ * @copyright Accredible <dev@accredible.com>
+ * @license   http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
+ */
+
+define(['jquery', 'core/ajax', 'core/templates'], function($, Ajax, Templates) {
+    let options = {};
+    const element = {
+        acrredibleSelect: '[id*="_accredibleattribute"]',
+        addButton: '[id*="_add_new_line"]',
+        list: '.attribute_mapping',
+    };
+    const optionsMap = {
+        coursefieldmapping: 'coursefieldoptions',
+        coursecustomfieldmapping: 'coursecustomfieldoptions',
+        userprofilefieldmapping: 'userprofilefieldoptions',
+    };
+
+    var mappings = {
```

Comment:
Question - Does `mappings` have to be mutable? Wouldn't it be better to use `const` here?

Reply:
No it doesn't have to be mutable, I've updated to `const`.